app/workers/postcodes_collection_worker.rb

```diff
@@ -0,0 +1,13 @@
+class PostcodesCollectionWorker < ApplicationWorker
+  sidekiq_options queue: :collect_postcodes
+
+  def perform
+    run_with_advisory_lock(Postcode, postcode) do
```

Comment:
The `sleep` I've suggested might be problematic with the `run_with_advisory_lock` block though. Reading the [docs](https://www.rubydoc.info/gems/with_advisory_lock/3.0.0), it implies to me that the database is locked until the lock is released, which in this case would take 12 million seconds! I think we're probably better off removing the `run_with_advisory_lock` method (unless I'm missing something?).